actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -145,6 +145,18 @@ def request_method
       @request_method ||= check_method(super)
     end
 
+    # Returns the URI pattern of the matched route for the request,
+    # using the same format as `bin/rails routes`:
+    #
+    #   request.route_uri_pattern # => "/:controller(/:action(/:id))(.:format)"
+    def route_uri_pattern
+      get_header("action_dispatch.route_uri_pattern")
+    end
+
+    def route_uri_pattern=(pattern) # :nodoc:
```

Comment:
@joelhawksley I've opened #47306 but I was thinking we might want to add a test for this if possible? WDYT?